### PR TITLE
Adding changes for configuring unsafe ssl for self signed certs use case

### PR DESCRIPTION
**Pull request by mayurvaid-redvest**
Signed-off-by: mayurvaid-redvest <109327850+mayurvaid-redvest@users.noreply.github.com>

_Provide a description of what has been changed_

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
- [x] Tests have been added
- [ ] A PR is opened to update our Helm chart ([repo](https://github.com/kedacore/charts)) *(if applicable, ie. when deployment manifests are modified)*
- [x] A PR is opened to update the documentation on ([repo](https://github.com/kedacore/keda-docs)) *(if applicable)*
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)

Fixes #

Relates to https://github.com/kedacore/keda/issues/2310

### CODE REVIEW

**Review by JorTurFer — COMMENTED** [review]
Looking good, could you add a unit test for this?

**Comment by mayurvaid-redvest** [comment]
Thanks @JorTurFer  for reviewing  , I have modified test case with new parameter and checked , Please let me know if anything else has to be added

**Comment by JorTurFer** [comment]
Yes, please open a PR in keda-doc adding the parameter to [Prometheus scaler page](https://github.com/kedacore/keda-docs/blob/main/content/docs/2.9/scalers/prometheus.md) and update the changelog. The issue you have to refeer is this https://github.com/kedacore/keda/issues/2310
And thanks a ton for the contribution!

**Comment by mayurvaid-redvest** [comment]
Docs Path : https://github.com/kedacore/keda-docs/pull/913

File: CHANGELOG.md [hunk 1]
```diff
@@ -41,7 +41,7 @@ To learn more about our roadmap, we recommend reading [this document](ROADMAP.md
 
 ### Improvements
 
-- TODO ([#XXX](https://github.com/kedacore/keda/issues/XXX))
+- **General:** Introduce skipping of certificate check for unsigned certs for Prometheus ([#2310](https://github.com/kedacore/keda/issues/2310))
```
Comment (JorTurFer) [inline CHANGELOG.md:44]:
```suggestion
- **Prometheus Scaler:** Introduce skipping of certificate check for unsigned certs ([#2310](https://github.com/kedacore/keda/issues/2310))
```

**Comment by mayurvaid-redvest** [comment]
@JorTurFer  , Can you please review and let me know anything pending

**Review by JorTurFer — APPROVED** [review]
LGTM!
Thanks for the contibution! ❤️

**Comment by JorTurFer** [comment]
Hi @mayurvaid-redvest ,
This change is ready to merge IMO, but we are planing to do a bugfix release soon (v2.8.1) and we don't introduce new features in that kind of releases, only fixes and security patches. This feature will be merged after the hotfix release and released as part of v2.9.0
Sorry for the inconvenience

**Comment by mayurvaid-redvest** [comment]
@JorTurFer Do i have any action now , Or just wait for change

**Comment by JorTurFer** [comment]
I think that you don't need to do any more. I'll let you know if we need something else
Thanks!

**Comment by mayurvaid-redvest** [comment]
> I think that you don't need to do any more. I'll let you know if we need something else Thanks!

Thanks for the help

**Comment by JorTurFer** [comment]
hey @mayurvaid-redvest 
Sorry but we need one thing more. After the release, there is a conflict with changelog file. Could you rebase main and solve the changelog please?

**Pull request closed without merging**